src/render_context.rs

```diff
@@ -58,6 +59,20 @@ impl<'a> RenderContext<'a> {
             }
         }
     }
+    /// return an empty rendering context
+    pub_unstable_internal! {
+        pub(crate) fn empty() -> Self {
+        let cached_set = &RefCell::new(CachedSet::default());
+        let bump = &Bump::new();
+        let templates = &mut FxHashMap::default();
+            RenderContext {
```

Comment:
I assume this is failing to compile because it is trying to return something that contains a borrow of something from the stack frame we are returning from?

Since `RenderContext` contains a borrow, but we don't want callers to have to supply any parameters that we will borrow from, I think we should make this a callback-based API:

```rust

pub(crate) fn empty<F, T>(f: F) -> T
where
    F: FnOnce(&mut RenderContext) -> T
{
    // ...
    f(&mut RenderContext { ... })
}

// and then usage would be like:

RenderContext::empty(|cx| {
    // use cx here...
})
```

Does that make sense?